src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -2398,6 +2418,36 @@ void grpc_chttp2_config_default_keepalive_args(grpc_channel_args *args,
                 &args->args[i],
                 (grpc_integer_options){g_default_keepalive_permit_without_calls,
                                        0, 1});
+      } else if (0 ==
+                 strcmp(args->args[i].key, GRPC_ARG_HTTP2_MAX_PING_STRIKES)) {
+        g_default_max_ping_strikes = grpc_channel_arg_get_integer(
```

Comment:
Thanks for the review!
These global values are not set from transport construction, they are set in `grpc_chttp2_config_default_keepalive_args()`. It's an internal interface, and must only be called at the initialization stage, as stated [here](https://github.com/grpc/grpc/blob/master/src/core/ext/transport/chttp2/transport/internal.h#L853).
The transport struct has corresponding properties in `grpc_chttp2_repeated_ping_policy`. When constructing a transport, we will use `g_default_max_ping_strikes` as the default value of `grpc_chttp2_transport.ping_policy.max_ping_strikes`.